### ISSUE TITLE
Brings back old read news items periodically.

### ISSUE THREAD

**Issue by hamishmb** [rh-id/a-news-provider#13]
News Provider tends to randomly start displaying old news items that I've already read from time to time. There doesn't seem to be a pattern to this, so I'm not really sure why.

**Comment by rh-id** [comment]
Not sure about this dont really see how it can happen unless you change the "filter by:" to none.

**Comment by hamishmb** [comment]
Yeah, nor me, but it does happen from time to time. I was wondering whether it had been fixed, but it happened just the other day. Sometimes the old news items have no title or content on the list.

**Comment by rh-id** [comment]
It could be caused by incorrect RSS record from the source. could you provide your RSS link?

**Comment by hamishmb** [comment]
Note: I'm using a Moto G8 Power with stock Android 11.

I have quite a few, but the main one is Phoronix: https://www.phoronix.com/rss.php

The no-content bug seems to be new actually, here are some screenshots of it (the following screenshots are what happens when I tap the no-content items).

![Screenshot_20221025-120103](https://user-images.githubusercontent.com/16725441/197849055-be627c0e-e4ab-4ab6-91fd-94898f2bacd1.png)

![Screenshot_20221025-120107](https://user-images.githubusercontent.com/16725441/197849075-12598957-ac03-49c6-a4ea-f4acb71ddf0e.png)

![Screenshot_20221025-120120](https://user-images.githubusercontent.com/16725441/197849096-a963a7c4-489b-4440-817a-3eee350981b6.png)

**Comment by rh-id** [comment]
Im not sure what caused this. I add the RSS link and I could see the title from the list. it seemed no issue on the RSS item records.

**Comment by rh-id** [comment]
Could you try to view it in landscape orientation? does it happen as well?

**Comment by hamishmb** [comment]
This seems to be unlikely to be a problem with Android itself, in my opinion - we'd be seeing things like this all the time if it were so. It seems feasible that it might be a bug in need of a workaround, though.

When it next occurs, I'll try switching to landscape mode and I'll take more screenshots.

I should note that if I close and re-open News Provider, the problem is usually gone when it re-opens.

**Comment by rh-id** [comment]
I think it was android problem because I dont see how it could happen inside the code. Also I personally use this app everyday and I never encounter this kind of strange issue even once, and I had tested this in oppo, samsung, and android emulator and it never happens.

**Comment by hamishmb** [comment]
That's very reasonable. I only rarely see the old items being brought back (maybe twice or three times in all my time using the app), and the blank items issue is fairly new. Perhaps I should file a separate issue for the blank items one? It seems a little easier to pin down.

**Comment by rh-id** [comment]
Just leave it open here. Perhaps other users that encounter the same issue can help adding more info.

**Comment by rh-id** [comment]
Just to update. I encounter this issue after I improve the app performance & memory. It seemed to inconsistently happened when you switch back to this app and the list is refreshed ( or when the list is refreshed ).

still not 100% sure why it can happen but Im guessing something happens when calculating the layout for RSS item, the latest release may fix this.

**Comment by rh-id** [comment]
I just fix many threading issue and possibly touching this Rss Item list as well, so we will close this first. check the latest release for the fix

**Comment by hamishmb** [comment]
Excellent. I can't see that release on F-droid yet - are you planning to add it there?

**Comment by hamishmb** [comment]
Still haven't been offered an update via F-droid, any reason why?
